src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IPropertyReferenceExpression.cs

```diff
@@ -45,5 +45,227 @@ class C2 : C1
+IPropertyReferenceOperation: System.Int32 C.I { get; } (OperationKind.PropertyReference, Type: System.Int32, IsInvalid) (Syntax: 'C.I')
+  Instance Receiver: 
+    null
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0120: An object reference is required for the non-static field, method, or property 'C.I'
+                //         var i1 = /*<bind>*/C.I/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_ObjectRequired, "C.I").WithArguments("C.I").WithLocation(8, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<MemberAccessExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IPropertyReference_StaticPropertyWithInstanceReceiver_Indexer()
+        {
+            string source = @"
+using System.Collections.Generic;
+
+class C
+{
+    static List<int> list = new List<int>();
+
+    public static void M()
+    {
+        var c = new C();
+        var i1 = /*<bind>*/c.list[1]/*</bind>*/;
+    }
+}
+";
+            string expectedOperationTree = @"
+IPropertyReferenceOperation: System.Int32 System.Collections.Generic.List<System.Int32>.this[System.Int32 index] { get; set; } (OperationKind.PropertyReference, Type: System.Int32, IsInvalid) (Syntax: 'c.list[1]')
+  Instance Receiver: 
+    IFieldReferenceOperation: System.Collections.Generic.List<System.Int32> C.list (Static) (OperationKind.FieldReference, Type: System.Collections.Generic.List<System.Int32>, IsInvalid) (Syntax: 'c.list')
+      Instance Receiver: 
+        ILocalReferenceOperation: c (OperationKind.LocalReference, Type: C, IsInvalid) (Syntax: 'c')
+  Arguments(1):
+      IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: index) (OperationKind.Argument, Type: System.Int32) (Syntax: '1')
+        ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+        InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+        OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // CS0176: Member 'C.list' cannot be accessed with an instance reference; qualify it with a type name instead
+                //         var i1 = /*<bind>*/c.list[1]/*</bind>*/;
+                Diagnostic(ErrorCode.ERR_ObjectProhibited, "c.list").WithArguments("C.list").WithLocation(11, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<ElementAccessExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact]
+        public void IPropertyReference_StaticPropertyAccessOnClass_Indexer()
```

Comment:
>IPropertyReference_StaticPropertyAccessOnClass_Indexer [](start = 20, length = 54)

It feels like the intent was to test access of indexer of off a type, but the test doesn't test that.

Reply:
Renamed this test, and added a new test for indexers.

---
In reply to: [149473156](https://github.com/dotnet/roslyn/pull/22972#discussion_r149473156) [](ancestors = 149473156)